python/src/otel/otel_sdk/otel_wrapper.py

```diff
@@ -48,8 +53,33 @@ def modify_module_name(module_name):
 class HandlerError(Exception):
     pass
 
-
-AwsLambdaInstrumentor().instrument()
+def _headers_and_sqs_context_extractor(lambda_event: Any) -> Context:
+    headers = {}
+    try:
+        headers = lambda_event["headers"]
+    except (TypeError, KeyError):
+        logger.debug(
+            "Extracting context from Lambda Event for headers failed."
+        )
+
+    sqs_ctx = {}
+    try:
+        records = lambda_event["Records"][0]['messageAttributes']
+        if 'traceparent' in records:
+            sqs_ctx['traceparent'] = records['traceparent']['stringValue']
+        if 'tracestate' in records:
+            sqs_ctx['tracestate'] = records['tracestate']['stringValue']
+        if 'baggage' in records:
+            sqs_ctx['baggage'] = records['baggage']['stringValue']
```

Comment:
This assumes the w3c propagator is configured. Can you convert the whole `records` object to something that can generically be passed into the extract so it can work with whichever propagator is configured?